lib/monitoring/uptime.ex

```diff
@@ -5,220 +5,147 @@ defmodule Monitoring.Uptime do
-
-        Logger.info([
-          "device_uptime: ",
-          "date_time=",
-          DateTime.to_string(date_time),
-          " device_type=dsp",
-          " line=",
-          line,
-          " station=",
-          String.replace(station, " ", "_"),
-          " dsp_id=",
-          dsp_id,
-          " is_online=",
-          is_online
-        ])
-
-      :amplifier ->
-        [line, station, amp_id | _] = String.split(description, ":")
-
-        Logger.info([
-          "device_uptime: ",
-          "date_time=",
-          DateTime.to_string(date_time),
-          " device_type=amp",
-          " line=",
-          line,
-          " station=",
-          String.replace(station, " ", "_"),
-          " amp_id=",
-          amp_id,
-          " is_online=",
-          is_online
-        ])
-
-      :comrex ->
-        [line, station, comrex_id | _] = String.split(description, ":")
-
-        Logger.info([
-          "device_uptime: ",
-          "date_time=",
-          DateTime.to_string(date_time),
-          " device_type=comrex",
-          " line=",
-          line,
-          " station=",
-          String.replace(station, " ", "_"),
-          " comrex_id=",
-          comrex_id,
-          " is_online=",
-          is_online
-        ])
+  defp get_device_status_fields(%{"description" => description, "is_online" => is_online} = node) do
+    device_type = get_device_type(node)
 
-      _ ->
-        Logger.info("unknown_node_type: #{inspect(node)}")
-    end
-  end
+    [line, station, device_id, maybe_sign_zone] = String.split(description, ":")
 
-  defp get_software_component_type(%{"sw_component" => sw_component}) do
-    case sw_component do
-      "Live PA" ->
-        :live_pa
+    log_fields = [
+      node_type: device_type,
+      line: line,
+      station: String.replace(station, " ", "_"),
+      device_id: device_id,
+      is_online: is_online
+    ]
 
-      _ ->
-        :unknown
-    end
+    if device_type == "sign",
+      do: log_fields ++ [sign_zone: maybe_sign_zone],
+      else: log_fields
   end
 
-  defp get_device_type(%{"node_type" => node_type} = _node) do
+  defp get_device_type(%{"node_type" => node_type}) do
     case node_type do
       "SGN" ->
-        :sign
+        "sign"
 
       "PSS" ->
-        :scu
+        "scu"
 
       "P8810" ->
-        :dsp
+        "dsp"
 
       "BLU-80" ->
-        :dsp
+        "dsp"
 
       "C4200" ->
-        :amplifier
+        "amplifier"
 
       "DGTY" ->
-        :comrex
+        "comrex"
 
       _ ->
-        :unknown
+        "unknown"
     end
   end
 
-  defp get_device_type(_), do: :unspecified
+  def log(_, _, []) do
+    nil
+  end
+
+  def log(prefix, date_time, fields) do
+    base = [
+      "#{prefix}: ",
+      "date_time=",
+      DateTime.to_string(date_time)
+    ]
+
+    Enum.reduce(fields, base, fn {k, v}, acc ->
+      acc ++ [" #{Atom.to_string(k)}=", v]
+    end)
+    |> Logger.info()
```

Comment:
This approach is good, but the code could be tuned up a bit:
* Logging via lists doesn't behave well if e.g. a number slips through, so it would be safer to send everything through interpolation.
* `date_time` could be prepended as a tuple to the field list, instead of being formatted separately.
* `reduce` could be replaced with a slightly more expressive `map` and `join`
* `Atom.to_string` is unnecessary if it's being interpolated